libraries/ui-library/src/components/six-tag/test/six-tag.spec.tsx

```diff
@@ -30,9 +41,11 @@ describe('six-tag', () => {
       <six-tag clearable="" size="medium" type="primary">
         <mock:shadow-root>
           <span class="tag tag--clear tag--medium tag--primary" part="base">
+          <six-tooltip>
             <span class="tag__content" part="content">
               <slot></slot>
             </span>
+          </six-tooltip>
```

Comment:
Hm, I guess this answer my question. Shouldn't we add the tooltip conditionally? In such a case like we are kind of unnecessarily polluting the ShadowDOM, no?

Reply:
It might be possible to do so, but that would cause the DOM to change (re-render) if the size of the tag changes (see resizeObserver in the implementation), which is probably not great performance wise.

Reply:
Hm but would it really be a performance issue? The render function would be anyway triggered, no?

I was just thinking of something like

```typescript
...
{ somecondition && <six-tooltip ref={(... }
```

Reply:
I am not a big fan of having a a tooltip wrapper element either for each tag element. But after trying out your suggestion, I prefer the wrapper approach, mostly because of the additional render and also because the tag flickers when the tooltip wrapper is added. The current implementation does not re-render on resize.

Also, my implementation of your suggestion felt a bit clunky, but it is very well possible I haven't found the most streamlined approach. I'm definitely open to an implementation suggestions if you have any.

Reply:
Ok fair enough, let's go with this for now, I mean flickering is definitely worse 😂